core/src/main/java/io/opencensus/stats/StatsManagerFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import io.opencensus.common.Clock;
+
+/**
+ * Object that determines which implementation of the {@link StatsManager} to use.
+ */
+public abstract class StatsManagerFactory {
+
+  /**
+   * Returns the default {@link StatsManager}.
+   */
+  public abstract StatsManager getDefaultStatsManager();
+
+  /**
+   * Creates a new {@link StatsManager} for testing.
+   */
+  public abstract StatsManager createTestStatsManager();
```

Comment:
> Do you think we shouldn't include the functionality, or is it just not necessary for gRPC?

I don't find it useful for gRPC, and I can't think of a potential use case. I would suggest not include this until a real use case comes up. It's always easier to add something than removing or changing it.

>  If we had an abstract StatsRecorder class that only had "record" methods on it, do you think that would be simple enough that user could mock it directly, and we wouldn't need to provide a testing implementation like you described?

Yes, as long as `StatsContext` loses its `record()` and becomes a pure data object.